### PR TITLE
Ingress service: make HAProxy to listen on IPv4 and IPv6 …

### PR DESCRIPTION
**Pull request by thegreenbear**
By default, HAProxy will only bind to IPv4 addresses preventing the ingress service to be used in dual stack or mixed IPv4/IPv6 environments. This patch changes the default so HAProxy would bind both on IPv4 and IPv6 addresses.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [X] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [X] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [X] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [X] No tests

### CODE REVIEW

**Comment by thegreenbear** [comment]
Tests are failing. Will fix.

**Comment by thegreenbear** [comment]
I think the tests failures (OSD building and RGW) are not related to my change.
What would be the next help?

**Comment by adk3798** [comment]
> I think the tests failures (OSD building and RGW) are not related to my change. What would be the next help?

if they're unrelated we can just trigger a rerun

**Comment by adk3798** [comment]
jenkins test make check

File: src/pybind/mgr/cephadm/tests/test_services.py [hunk 1]
```diff
@@ -2062,7 +2062,7 @@ def test_ingress_config_multi_vips(self, _run_cephadm, cephadm_module: CephadmOr
                                 'http-request use-service prometheus-exporter if { path /metrics }\n    '
                                 'monitor-uri /health\n'
                                 '\nfrontend frontend\n    '
-                                'bind *:8089\n    '
+                                'bind [..]:8089\n    '
```
Comment (adk3798) [inline src/pybind/mgr/cephadm/tests/test_services.py:2065]: A few lines above this, there is another `bind *:8999\n` line that needs the same adjustment. Currently the unit test fails
```
E                 Full diff:
E                   {
E                    'files': {'haproxy.cfg': '# This file is generated by cephadm.\n'
E                                             'global\n'
E                                             '    log         127.0.0.1 local2\n'
E                                             '    chroot      /var/lib/haproxy\n'
E                                             '    pidfile     /var/lib/haproxy/haproxy.pid\n'
E                                             '    maxconn     8000\n'
E                                             '    daemon\n'
E                                             '    stats socket /var/lib/haproxy/stats\n'
E                                             '\n'
E                                             'defaults\n'
E                                             '    mode                    http\n'
E                                             '    log                     global\n'
E                                             '    option                  httplog\n'
E                                             '    option                  dontlognull\n'
E                                             '    option http-server-close\n'
E                                             '    option forwardfor       except 127.0.0.0/8\n'
E                                             '    option                  redispatch\n'
E                                             '    retries                 3\n'
E                                             '    timeout queue           20s\n'
E                                             '    timeout connect         5s\n'
E                                             '    timeout http-request    1s\n'
E                                             '    timeout http-keep-alive 5s\n'
E                                             '    timeout client          30s\n'
E                                             '    timeout server          30s\n'
E                                             '    timeout check           5s\n'
E                                             '    maxconn                 8000\n'
E                                             '\n'
E                                             'frontend stats\n'
E                                             '    mode http\n'
E                 -                           '    bind *:8999\n'
E                 ?                                     ^
E                 +                           '    bind [..]:8999\n'
E                 ?                                     ^^^^
E                                             '    bind 1.2.3.7:8999\n'
E                                             '    stats enable\n'
E                                             '    stats uri /stats\n'
E                                             '    stats refresh 10s\n'
E                                             '    stats auth admin:12345\n'
E                                             '    http-request use-service prometheus-exporter if '
E                                             '{ path /metrics }\n'
E                                             '    monitor-uri /health\n'
E                                             '\n'
E                                             'frontend frontend\n'
E                                             '    bind [..]:8089\n'
E                                             '    default_backend backend\n'
E                                             '\n'
E                                             'backend backend\n'
E                                             '    option forwardfor\n'
E                                             '    balance static-rr\n'
E                                             '    option httpchk HEAD / HTTP/1.0\n'
E                                             '    server rgw.foo.test.gwznjd 1.2.3.7:80 check '
E                                             'weight 100\n'},
E                   }

```

**Review by adk3798 — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/cephadm/tests/test_services.py (hunk 1 above)
Comment (thegreenbear) [reply src/pybind/mgr/cephadm/tests/test_services.py:2065]: Oops. Allright, should be fixed.
Let's see if the test pass this time. Thanks for the comment.

**Comment by thegreenbear** [comment]
Ok, now I only see errors related to the RGW. So all's good? Or am I reading this wrong?

**Review by adk3798 — APPROVED** [review]

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-03-16_05:03:17-orch:cephadm-wip-adk-testing-2024-03-15-1513-distro-default-smithi/

mostly cluster log stuff + mds_upgrade_sequence. Other failures were either known or not reproducible.

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-03-16_05:03:17-orch:cephadm-wip-adk-testing-2024-03-15-1513-distro-default-smithi/

Tons of cluster log failures, not a problem for now. Other than that, it was just the mds_upgrade_sequence, staggered upgrade with agent (both of which are known problems), some failures installing things at the beginning of tests (known to happen occasionally), and a failure in the test_cephadm test related to idmap overrides for nfs, which I missed when we tested that PR originally `ERROR: required file missing from config-json: idmap.conf` but is not being caused by any PRs in this run.

Nothing to block merging PRs from this run.

**Pull request merged**